packages/pwa-kit-dev/src/configs/webpack/config.js

```diff
@@ -212,9 +218,14 @@ const withChunking = (config) => {
     }
 }
 
+const addReactRefresh = (babelPlugins) =>
+    [...babelPlugins, mode === development && require.resolve('react-refresh/babel')].filter(
+        Boolean
+    )
+
 const client =
     entryPointExists(['app', 'main']) &&
-    baseConfig('web')
+    baseConfig('web', {addBabelPlugins: addReactRefresh})
```

Comment:
have you considered extending the `Builder` class to add class methods like `enableReactRefresh`.

```js
class Builder{
  constructor() {...}
  extend() {...}
  build() {...}

  // new method
  enableReactRefresh() {}
}

Reply:
actually, the builder class already allows customization thru the `extend` method...

i think you can do something like

```js
baseConfig('web').extend(enableReactRefresh)
```

Reply:
but i personally like the explicit class method approach more

Reply:
if you do choose to go with the explicit class method, let's update `baseConfig('web').extend(withChunking)` to `baseConfig('web').withChunking()`, so the code is consistent.